addons/controller_icons/ControllerIcons.gd

```diff
@@ -164,12 +171,15 @@ func _add_custom_input_action(input_action: String, data: Dictionary):
 
 func refresh():
 	# All it takes is to signal icons to refresh paths
-	emit_signal("input_type_changed", _last_input_type)
+	emit_signal("input_type_changed", _last_input_type, _last_controller)
+
+func get_joypad_type(controller: int = _last_controller) -> ControllerSettings.Devices:
```

Comment:
Is is necessary to expose this? Can you share what's your use-case for this?

Reply:
Right, it isn't related to this PR but I forgot to remove it. My usecase is to detect switch controllers to give them a different button mapping.

Please let me know if you prefer to remove it, keep it or get a separate PR.

Reply:
I assumed there was a valid reason for this, just wanted a confirmation that this was intended and not leftovers.

Your use case is valid, and not having this would make for some nasty code that requires access to private fields/methods, so I think it's good to stay :+1: